substratevm/src/com.oracle.graal.pointsto.standalone/src/com/oracle/graal/pointsto/standalone/features/StandaloneClassInitializationFeature.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2022, 2022, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2022, 2022, Alibaba Group Holding Limited. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.oracle.graal.pointsto.standalone.features;
+
+import com.oracle.graal.pointsto.meta.AnalysisMethod;
+import com.oracle.graal.pointsto.meta.AnalysisType;
+import org.graalvm.nativeimage.hosted.Feature;
+
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class StandaloneClassInitializationFeature implements Feature {
+
+    Set<AnalysisMethod> addedClinits = ConcurrentHashMap.newKeySet();
+
+    @Override
+    public void duringAnalysis(DuringAnalysisAccess a) {
+        StandaloneAnalysisFeatureImpl.DuringAnalysisAccessImpl access = (StandaloneAnalysisFeatureImpl.DuringAnalysisAccessImpl) a;
+        for (AnalysisType type : access.getUniverse().getTypes()) {
+            if (type.isReachable()) {
```

Comment:
You could avoid using a feature here by registering the initializer as a root method in `com.oracle.graal.pointsto.standalone.StandaloneHost.initializeType()`. This is called from the `com.oracle.graal.pointsto.meta.AnalysisType.initializationTask` when the type is marked as reachable.